packages/trpc/server/routers/viewer/bookings/prevResponse.handler.ts

```diff
@@ -0,0 +1,31 @@
+import type { PrismaClient } from "@calcom/prisma";
+
+import type { TrpcSessionUser } from "../../../trpc";
+import type { TPrevResponseInputSchema } from "./prevResponse.schema";
+
+type PrevResponseOptions = {
+  ctx: {
+    user: NonNullable<TrpcSessionUser>;
+    prisma: PrismaClient;
+  };
+  input: TPrevResponseInputSchema;
+};
+
+export const prevResponseHandler = async ({ ctx, input }: PrevResponseOptions) => {
+  const { eventTypeId } = input;
+  const { user, prisma } = ctx;
+
+  return await prisma.booking.findFirst({
+    where: {
+      eventTypeId,
+      attendees: {
+        some: {
+          email: user.email,
+        },
+      },
+    },
```

Comment:
This has a potential of being problematic if 'Booker email verification' isn't enabled. Booking details submitted by spammer/hacker can appear in the real person's booking page auto-filled.

Reply:
Also, after let's say we ensure that 'Booker Email verification' is enabled, the query would allow the wrong data to be prefilled.  If a user adds a guest email guest@example.com during a booking and let's say later on that guest does a booking himself, it would wrongly prefill the booking response. 

We need to work with booker's data and not attendee's.

Reply:
One possible way to avoid the two problems above is to maybe store the last response locally in browser and use that only.

Reply:
Yeah I didn't think of this. Storing it locally would make the most sense and would work for bookers who haven't signed up.